packages/property-grid/src/components/PresentationPropertyGridWidget2.tsx

```diff
@@ -0,0 +1,772 @@
+    async (propName: string) => {
+      if (!projectId || !sharedFavorites) {
+        setContextMenu(undefined);
+        return;
+      }
+      sharedFavorites.push(propName);
+
+      const requestContext = await AuthorizedFrontendRequestContext.create();
+      const result = await IModelApp.settings.saveSharedSetting(
+        requestContext,
+        sharedFavorites,
+        sharedNamespace,
+        sharedName,
+        false,
+        projectId,
+        iModelConnection?.iModelId
+      );
+      if (result.status !== SettingsStatus.Success) {
+        throw new Error(
+          "Could not share favoriteProperties: " + result.errorMessage
+        );
+      }
+      const result2 = await IModelApp.settings.getSharedSetting(
+        requestContext,
+        sharedNamespace,
+        sharedName,
+        false,
+        projectId,
+        iModelConnection?.iModelId
+      );
+      if (result2.status !== SettingsStatus.Success) {
+        throw new Error(
+          "Could not share favoriteProperties: " + result2.errorMessage
+        );
+      }
+      setContextMenu(undefined);
+    },
+    [iModelConnection?.iModelId, projectId, sharedFavorites]
+  );
+
+  const onUnshareFavorite = React.useCallback(
+    async (propName: string) => {
+      if (!projectId || !sharedFavorites) {
+        setContextMenu(undefined);
+        return;
+      }
+      const index = sharedFavorites.indexOf(propName);
+      if (index > -1) {
+        sharedFavorites.splice(index, 1);
+      }
+      const requestContext = await AuthorizedFrontendRequestContext.create();
+      const result = await IModelApp.settings.saveSharedSetting(
+        requestContext,
+        sharedFavorites,
+        sharedNamespace,
+        sharedName,
+        false,
+        projectId,
+        iModelConnection?.iModelId
+      );
+      if (result.status !== SettingsStatus.Success) {
+        throw new Error(
+          "Could not unshare favoriteProperties: " + result.errorMessage
+        );
+      }
+      setContextMenu(undefined);
+    },
+    [iModelConnection?.iModelId, projectId, sharedFavorites]
+  );
+
+  const shareActionButtonRenderer: ActionButtonRenderer = (
+    _props: ActionButtonRendererProps
+  ) => {
+    const shared =
+      sharedFavorites !== undefined &&
+      sharedFavorites?.findIndex(
+        (fav: string) => _props.property.property.name === fav
+      ) >= 0;
+    return (
+      <div>
+        {shared && (
+          <span
+            className="icon icon-share"
+            style={{ paddingRight: "5px" }}
+          ></span>
+        )}
+      </div>
+    );
+  };
+
+  const onCopyText = React.useCallback(
+    async (property: PropertyRecord) => {
+      if (property.description) copyToClipboard(property.description);
+      else if (props.debugLog)
+        props.debugLog(
+          "PROPERTIES COPY TEXT FAILED TO RUN DUE TO UNDEFINED PROPERTY RECORD DESCRIPTION"
+        );
+      setContextMenu(undefined);
+    },
+    [props]
```

Comment:
i would much rathe ryou destruct all props, and just pass in what you need into the dependency list
will also be easier to figure out what props arent being used